core/src/test/java/org/apache/druid/java/util/common/parsers/TimestampParserTest.java

```diff
@@ -208,8 +208,10 @@ public void testFormatsForNumberBasedTimestamp()
     expectedDt = DateTimes.of("2020-05-15T20:20:40.000Z");
 
     parser = TimestampParser.createObjectTimestampParser("millis");
-    Assert.assertEquals("Timestamp of format millis not parsed correctly",
+    Assert.assertEquals("Timestamp of format millis as long not parsed correctly",
                         expectedDt, parser.apply(millis));
+    Assert.assertEquals("Timestamp of format millis as string not parsed correctly",
```

Comment:
@suneet-s - this test case passes fine without the fix in #9877. #9877 was meant for use cases where the ingested column storing time is a numeric column (int, long) but it stores time in that isn't `auto || millis || posix || micro || nano || ruby`

Reply:
@samarthjain makes sense. I initially thought this was a missing test case before your pr. Looking more closely at the tests, I see this is covered by `TimestampParserTest#testNano`

Closing this PR.